numba/np/arrayobj.py

```diff
@@ -4545,21 +4531,28 @@ def impl(start, stop=None, step=None, dtype=None):
     return impl
 
 
-@glue_lowering(np.linspace, types.Number, types.Number)
-def numpy_linspace_2(context, builder, sig, args):
+@overload(np.linspace)
+def numpy_linspace_2(start, stop):
+    if isinstance(start, types.Number) and isinstance(stop, types.Number):
+        def impl(start, stop):
+            return np.linspace(start, stop, 50)
+        return impl
```

Comment:
I don't think that this version should exist. The default value for `num` should be set as a kwarg in the 3 arg variant, like `start, stop, num=50`.